### PR TITLE
Markdown: sync expected test results and input data with problem-specifications.

### PR DESCRIPTION
**Pull request by BethanyG**
Part of https://github.com/exercism/python/issues/1762

**NOTE:**  `flake8` will report the following for the exercise stub (in this case it isn't really a *stub*, but more a file to refactor):  _**"C901 'parse' is too complex (19)**_".  Which is the point : the file needs to be refactored to reduce complexity.

- Changed  function name to **parse** in tests, `example.py`, and exercise input to conform with `canonical-data.json`.
- Changed abbreviated variable name **res** to **result** in `example.py`, for clarity.
- Changed other single-letter variable names to words in `example.py`, for clarity.

### CODE REVIEW

**Comment by BethanyG** [comment]
arg.  apologies.  Forgot to update this branch.  Will update for those that follow.

**Comment by cmccandless** [comment]
> NOTE: flake8 will report the following for the exercise stub (in this case it isn't really a stub, but more a file to refactor): "C901 'parse' is too complex (19)".

`flake8` passed in the Travis builds. You might have a non-default configuration for your `flake8` installation.

File: exercises/markdown/example.py [hunk 1]
```diff
@@ -22,9 +22,9 @@ def wrap(line, tag):
 
 def check_headers(line):
     pattern = '# (.*)'
-    for i in range(6):
+    for index in range(6):
```
Comment (cmccandless) [inline exercises/markdown/example.py]: Out of curiosity: does `flake8` complain about this one, or was that your personal style preference?

**Comment by cmccandless** [comment]
Merged; thanks for working on this!

File: exercises/markdown/example.py (hunk 1 above)
Comment (BethanyG) [reply exercises/markdown/example.py]: Apologies - for whatever reason, this comment did not get emailed to me...so I am only seeing it now.  That change is a personal preference.  I can always refrain if it is bugging you/not good practice in this situation.

Comment (cmccandless) [reply exercises/markdown/example.py]: Doesn't bother me; I just see `for i in range(n)` so frequently that I generally give that one a pass on variable naming rule.

**Pull request merged**